test/unit/foundation/schema/attribute.test.ts

```diff
@@ -0,0 +1,386 @@
+import { expect } from '@open-wc/testing';
+
+import { getSchema } from '../../../../src/schemas.js';
+import { schemaAttributeDefinition } from '../../../../src/foundation.js';
+
+const schema2007B4: XMLDocument = <XMLDocument>(
+  new DOMParser().parseFromString(
+    getSchema('2007', 'B', '4'),
+    'application/xml'
+  )
+);
+
+const schema2007B: XMLDocument = <XMLDocument>(
+  new DOMParser().parseFromString(getSchema('2007', 'B', ''), 'application/xml')
+);
+
+const schema2003: XMLDocument = <XMLDocument>(
+  new DOMParser().parseFromString(getSchema('', '', ''), 'application/xml')
+);
+
+describe('schema spcific functions', () => {
+  describe('allow to get attribute definition from schema that', () => {
+    it('returns empty array for invalid tagName', () =>
+      expect(schemaAttributeDefinition('nonsence', schema2007B4)).to.be.empty);
+
+    describe('for element SampledValueControl', () => {
+      describe('and version 2003', () => {
+        const attributes = schemaAttributeDefinition(
+          'SampledValueControl',
+          schema2003
+        );
+
+        it('return the correct number of attributes ', () =>
```

Comment:
There is a few of these with a space at the end you might like to remove.

```suggestion
        it('return the correct number of attributes', () =>
```